include/eve/algo/container/soa_vector.hpp

```diff
@@ -0,0 +1,277 @@
+    //! @param v Value to write
+    EVE_FORCEINLINE void set(std::size_t i, value_type const& v)
+    {
+      return eve::write(begin() + i, kumi::flatten_all(v));
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Iterators
+    //! @{
+    //==============================================================================================
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto begin_aligned()  { return begin_aligned_impl(*this); }
+
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto begin_aligned()  const { return begin_aligned_impl(*this); }
+
+    //! Returns a constant aligned iterator to the beginning
+    EVE_FORCEINLINE auto aligned_cbegin() const { return begin_aligned(); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  { return eve::algo::unalign(begin_aligned()); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  const { return eve::algo::unalign(begin_aligned()); }
+
+    //! Returns a constant iterator to the beginning
+    EVE_FORCEINLINE auto cbegin() const { return begin(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  { return begin() + size(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  const { return begin() + size(); }
+
+    //! Returns a constant iterator to the end
+    EVE_FORCEINLINE auto cend() const { return end(); }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Comparisons and ordering
+    //! @{
+    //==============================================================================================
+    friend auto operator<=>(soa_vector const& lhs, soa_vector const& rhs) = default;
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    friend std::ostream& operator<<(std::ostream& os, soa_vector const& v)
+    {
+      os << "{ ";
+      for(std::size_t i=0;i<v.size();++i)
+        os << v.get(i) << " ";
+      os << '}';
+      return os;
+    }
+
+    template <typename Traits, typename Self>
+    requires std::same_as<std::remove_reference_t<Self>, soa_vector>
+    EVE_FORCEINLINE friend auto tagged_dispatch(algo::preprocess_range_, Traits tr, Self& self)
```

Comment:
I don't think you tested this for `const`.

Reply:
I don't think you tested any algos and fix it when I do convert here, np.
I can do that, np.